### DISCUSSION TITLE
How to dynamically update items (lines) in OptionList?

### DISCUSSION THREAD

**Discussion by StanSvec** [Textualize/textual#2241] [Q&A] [answered]
I need to dynamically display processes in a tabular format (like the ps command) in a live view using `OptionList` as a workaround for missing `remove_row` method of `DataTable`.  The idea is to use a custom **Rich** component with a `__rich__` method that returns a `rich.text.Text` instance for individual columns in row. Not sure if that's the best idea, but it may possibly work.

The only issue is that I cannot make the `OptionList` to redraw changed lines when there is an update in a column. Nothing is refreshed when `OptionList.refresh()` method is executed. However when a new line (option) is added/removed it refreshes everything including changed columns of the existing lines.

Any idea please how to force the refresh? There is also a `refresh` method accepting a region as a parameter. Perhaps there is a way to calculate the region for a line?

**Comment by davep** [comment, marked as answer]
The issue here is that the `OptionList` isn't built to display items that have their underlying data changing on the fly; it's designed to display a list of values passed to it. By the sounds of it it does seem like the `DataTable` would be a better choice for you here; but if you need the updated version that allows for row removal and you can't wait, perhaps making a quick custom widget based off the [line API](https://textual.textualize.io/guide/widgets/#line-api) would be the best choice for you here?

**Reply by StanSvec** [reply]
Thank you for the explanation. I'm not in a rush, so I'll just wait for the removal to become available in the data table.
